vcr/issuer/oidc4vci_issuer.go

```diff
@@ -184,6 +195,106 @@ func (i *memoryIssuer) HandleCredentialRequest(ctx context.Context, issuer did.D
 	return &credential, nil
 }
 
+func (i *memoryIssuer) validateProof(request oidc4vci.CredentialRequest, issuer did.DID, wallet did.DID) error {
+	if request.Proof == nil {
+		return oidc4vci.Error{
+			Err:        errors.New("missing proof"),
+			Code:       oidc4vci.InvalidOrMissingProof,
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+	if request.Proof.ProofType != oidc4vci.ProofTypeJWT {
+		return oidc4vci.Error{
+			Err:        errors.New("proof type not supported"),
+			Code:       oidc4vci.InvalidOrMissingProof,
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+	token, err := crypto.ParseJWT(request.Proof.Jwt, func(kid string) (crypt.PublicKey, error) {
```

Comment:
could it be possible to use a standard keyResolver function and place the signerDID==wallet out of the function?